src/components/forms/TextField/TextField.tsx

```diff
@@ -81,11 +89,13 @@ export const TextField: React.FC<TextFieldProps> = props => {
     }
   }, [value])
 
+  const inputType = dataType === 'number' ? 'text' : dataType
```

Comment:
This is bad for usability on mobile devices but we have to do like that until we improve the validity system. Currently having `number` as input type prevent us to display the corresponding error message...